### ISSUE TITLE
Add transformer, llm, ML and graph-rl matching processes

### ISSUE THREAD

**Issue by matentzn** [mapping-commons/semantic-mapping-vocabulary#27]
Due to some work with @cassiatrojahn and @sven-h I am suggesting to add the following categories of match types related to neural networks in the wider sense.

IRI | skos:prefLabel | skos:definition | dc:source | skos:example | rdfs:comment | altLabel | Parent
--- | --- | --- | --- | --- | --- | --- | ---
semapv:TransformerBasedMatching | transformer-based matching process | A matching process that utilizes transformer models, which are a type of deep learning model architecture designed to handle sequential data, particularly for natural language processing tasks. |  | Matches between entities are established based on the contextual relationships learned by the transformer from large datasets. |  |  | semapv:Matching
semapv:LLMBasedMatching | LLM-based matching process | A matching process that employs large language models (LLMs) which are pre-trained on vast amounts of text data and can understand and generate human-like text, making them suitable for tasks requiring a deep understanding of language. |  | Matches between entities are determined through the language understanding capabilities of LLMs, such as semantic context and language inference. |  |  | semapv:Matching
semapv:MachineLearningBasedMatching | machine learning-based matching process | A matching process that involves machine learning algorithms which learn from data to find patterns or make decisions with minimal human intervention. |  | Matches between entities are made by applying learned models to data points to predict similarities or relationships. |  |  | semapv:Matching
semapv:GraphRepresentationLearningBasedMatching | graph representation learning-based matching process | A matching process that uses graph representation learning which is a method in machine learning that focuses on learning a compact representation for graphs, capturing their structural information. |  | Matches between entities are identified by analyzing the learned representations that encode the structural features and relationships within graph data. |  |  | semapv:Matching

**Comment by noelmcloughlin** [comment]
Inge (AI Atlas) mentioned this work to me.  I guess `semapv:LLMBasedMatching` is not official yet.

**Comment by matentzn** [comment]
@sven-h @cassiatrojahn — went ahead with #29 but reshuffled the original proposal a bit, want your sanity check before merging.

The four classes you proposed were all flat siblings under `Matching`. I made `MachineLearningBasedMatching` the parent of the others, and introduced an intermediate class `EmbeddingBasedMatching` to group `TransformerBasedMatching` and `GraphRepresentationLearningBasedMatching`:

```
Matching
└── MachineLearningBasedMatching
    ├── EmbeddingBasedMatching
    │   ├── TransformerBasedMatching
    │   └── GraphRepresentationLearningBasedMatching
    └── LLMBasedMatching
```

The axis I'm splitting on is *how the model is used*, not *what kind of model*: embedding-based is "compare learned representations under a similarity measure", LLM-based is "the model decides directly via prompting or fine-tuning". That keeps transformer and LLM disjoint in practice even though current LLMs *are* transformers — the discriminator is the role the model plays, not its architecture.

I also added `AgentBasedMatching` as a separate direct child of `Matching` (not under `LLMBasedMatching`) — extending the original scope a bit. The distinguishing feature there is the workflow shape (autonomous controller deciding at runtime what tools to call, what to retrieve, when to stop), not the model underneath. Subclassing under `LLMBasedMatching` would be an OWL universal claim that every agentic matcher is LLM-based, which is true today but brittle (symbolic planners, RL agents, neurosymbolic etc.). A concrete process can be both `AgentBasedMatching` and `LLMBasedMatching` — they're orthogonal, not disjoint.

Last change: I rewrote the definitions to describe the matching approach rather than the underlying tech — e.g. "uses transformer models to produce contextual vector representations of entity features for similarity comparison" instead of "transformer models are a deep learning architecture designed to handle sequential data". The textbook background lives in `rdfs:comment`.

@noelmcloughlin — to your question above: not official yet, but #29 is the answer once it merges.

**Comment by sven-h** [comment]
Hi @matentzn ,
if you want to split based on "how the model is used", do you want to introduce a "Blocking" / "Candidate Generation" class (which correponds to "compare learned representations under a similarity measure")? But this is completely different from the matchers (which corresponds to "the model decides directly via prompting or fine-tuning"). Blockers generate a lot of correspondences (with high recall), whereas matchers analyze those and decide which ones are correct.
Both types can use transformed-based methods or not.

The key difference is that during blocking you only encode one entity (without knowing other entities) whereas during matching / reranking, you know already the candidates (source and target) and can do a lot more comparison approaches etc.

I don't know about the others but I would still keep the LLM under transformer. 
There might be LLMs in the future that do not use the transformer architecture, but then there might be a different name (like xLSTM or LMMs - Large Memory Models).

The agent based one if fine for me.

Final structure:
```
Blocking
└── EmbeddingBasedBlocking
Matching
└── MachineLearningBasedMatching
    ├── TransformerBasedMatching
    │      └── LLMBasedMatching
    └── GraphRepresentationLearningBasedMatching
    └── AgentBasedMatching 
```

**Comment by cthoyt** [comment]
@sven-h do you have references to concrete examples where these ideas have been implemented / described?

**Comment by sven-h** [comment]
- EmbeddingBasedBlocking (all approaches that makes use of sentence transformer) 
    - https://art.uniroma2.it/publications/docs/2025_SEMANTICS25_SISMA.pdf
    - https://arxiv.org/abs/2311.03837
    - https://arxiv.org/pdf/2508.10703
    - https://ceur-ws.org/Vol-4144/om2025-LTpaper4.pdf
    - also many more
- TransformerBasedMatching (I would count all approaches that uses e.g. BERT)
    - https://arxiv.org/html/2310.11244v2
    - https://arxiv.org/abs/2112.02682
    - https://arxiv.org/abs/2202.09791
    - https://dl.acm.org/doi/10.14778/3467861.3467878
    - https://www.vldb.org/pvldb/vol15/p1726-paganelli.pdf
 - LLMBasedMatching
     - https://arxiv.org/abs/2507.14032
     - https://arxiv.org/abs/2311.03837
     - https://arxiv.org/abs/2501.11441
     - https://arxiv.org/abs/2404.10317
     - https://ceur-ws.org/Vol-4144/om2025-LTpaper3.pdf
     - https://ceur-ws.org/Vol-4144/om2025-LTpaper5.pdf
     - many more
- AgentBasedMatching 
     - https://www.vldb.org/pvldb/vol18/p516-qiang.pdf
     - more in the future, I think

**Comment by sven-h** [comment]
In general, the ontology matching workshop contains a lot of such systems:
- 2025: https://ceur-ws.org/Vol-4144/
- 2024: https://ceur-ws.org/Vol-3897/

And Hamed also implements many of those systems in [ontoaligner](https://ontoaligner.readthedocs.io/) ([paper](https://arxiv.org/abs/2503.21902))

**Comment by matentzn** [comment]
Thanks @sven-h, @cthoyt. Quick summary of what changed (some pushing back, sorry both).

**Implemented**

- @sven-h `Blocking` + `EmbeddingBasedBlocking`: added under `MappingActivity` (Blocking defined by pipeline role, see pushback below).
- @sven-h's reference list: used as `dc:source` on the new classes, preferring DOIs where I could find them.
- @sven-h `AgentBasedMatching`: kept where it was (direct child of `Matching`); your diagram was ambiguous, "fine for me" read as approving the placement.
- @cthoyt: every new class now has a concrete `dc:source` and a software/example mention (BERTMap, OLaLa, OntoAligner, etc.).
- @cthoyt: `BoundedPathMatching` got a concrete example and the Euzenat & Shvaiko citation.

**Pushing back gently**

- @sven-h on "embedding similarity = blocking". I see your point but I personally think this conflates two orthogonal axes: pipeline role (candidate generation vs decision) and encoding style (independent vs joint). A sentence-BERT + cosine-threshold system with no reranker is a complete matcher; calling it "blocking with no matcher" contradicts SEMAPV's current definition of `Matching` ("a process that results in a mapping"). So I kept `EmbeddingBasedMatching` for the "similarity IS the decision" case and added `EmbeddingBasedBlocking` for the candidate-generation case. SSSOM does not have a good guideline for documenting blocking yet, but this is a separate matter.
- @sven-h on `LLMBasedMatching ⊑ TransformerBasedMatching`. I'd rather keep them as siblings. The discriminator that matters for provenance is the *role* the model plays (embedding generator vs decision-maker), not its architecture. Subclassing LLM under Transformer is an OWL universal claim that breaks the day a non-transformer LLM (Mamba, RWKV, xLSTM) is used and still called an LLM; I do also agree that Large Language Model with a captial L is universally associated with the transformer architecture, but large (lower-case, if you will) as a descriptor not necessarily (I get your point about that "they will be called something else" - but that is not certain, so I'd rather stay a bit uncommitted here).
- @cthoyt on "graph machine learning": kept `GraphRepresentationLearningBasedMatching` as the prefLabel; "graph representation learning" is the standard ML term (Hamilton's textbook of that name, GraphSAGE, node2vec, etc.). Added "graph machine learning-based matching" as `altLabel` so the broader term still resolves.

Open to revisiting any of this, please continue pushing back - building ontologies is never easy :)

THANKS A TON BOTH!

**Comment by sven-h** [comment]
Fine for me :+1:
